homeassistant/components/hdmi_cec/__init__.py

```diff
@@ -210,6 +214,18 @@ def setup(hass: HomeAssistant, base_config):
         adapter = CecAdapter(name=display_name[:12], activate_source=False)
     hdmi_network = HDMINetwork(adapter, loop=loop)
 
+    def _adapter_watchdog(now=None):
+        _LOGGER.debug("Reached _adapter_watchdog")
+        event.async_call_later(hass, WATCHDOG_INTERVAL, _adapter_watchdog)
+        if not adapter.initialized:
+            _LOGGER.info("Adapter not initialized. Trying to restart.")
+            hass.bus.fire(EVENT_HDMI_CEC_UNAVAILABLE)
+            adapter.init()
+
+    hdmi_network.set_initialized_callback(
+        partial(event.async_call_later, hass, WATCHDOG_INTERVAL, _adapter_watchdog)
```

Comment:
Will this callback be called from the same event loop as Home Assistant is using?